File: 3st/Juhee/콜라츠 수열 만들기.java
class Solution {
    public int[] solution(int n) {
        int[] temp = new int[1000];
        int index = 0;
        
        temp[index++] = n;

        while (n != 1) {
            if (n % 2 == 0) {
                n /= 2;
            } 
            else {
                n = 3 * n + 1;
            }
            temp[index++] = n;
        }
        int[] answer = new int[index];
        for (int i = 0; i < index; i++) {
            answer[i] = temp[i];
        }
        return answer;
    }
}

File: 3st/Hyeonseo/programmer/콜라츠 수열 만들기.py
def solution(n):
    answer = [n]   # 처음 시작값 넣기
    
    while n != 1:  # 1이 될 때까지 반복
        if n % 2 == 0:   # 짝수라면
            n = n // 2
        else:            # 홀수라면
            n = 3 * n + 1
        answer.append(n) # 다음 값 추가
    
    return answer
